From c4a79dd1a3397eb316fc126714482cad7da869ff Mon Sep 17 00:00:00 2001
From: Katsuya Matsubara <matsu@igel.co.jp>
Date: Tue, 5 Jun 2012 14:14:25 +0900
Subject: [PATCH 50/73] rcarvin: adjust buffer size to multiple of 128 bytes

VIN requires that buffer address must be aligned with 128 bytes.
In addition, address of UV plane for NV16 must be aligned with it too.
This size adjustment may induce the address alignment for each buffer
and UV plane.
---
 drivers/media/video/rcarvin.c |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/drivers/media/video/rcarvin.c b/drivers/media/video/rcarvin.c
index cfc27c3..bb03ce4 100644
--- a/drivers/media/video/rcarvin.c
+++ b/drivers/media/video/rcarvin.c
@@ -271,11 +271,16 @@ static int rcar_vin_videobuf_setup(struct videobuf_queue *vq,
 	struct rcar_vin_dev *pcdev = ici->priv;
 	int bytes_per_line = soc_mbus_bytes_per_line(icd->user_width,
 						icd->current_fmt->host_fmt);
+	unsigned int align;
 
 	if (bytes_per_line < 0)
 		return bytes_per_line;
 
-	*size = bytes_per_line * icd->user_height;
+	/* VIN requires 128 bytes aligned address
+	   for each buffer and plane. */
+	align = (icd->current_fmt->host_fmt->
+		 fourcc == V4L2_PIX_FMT_NV16) ? 0xff : 0x7f;
+	*size = ((bytes_per_line * icd->user_height) + align) & ~align;
 
 	if (0 == *count)
 		*count = 2;
-- 
1.7.0.4

